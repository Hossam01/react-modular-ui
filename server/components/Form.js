'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _AbstractComponent = require('./AbstractComponent');

var _AbstractComponent2 = _interopRequireDefault(_AbstractComponent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

class Form extends _AbstractComponent2.default {
  onChange(e) {
    if (typeof this.props.onChange === 'function') {
      this.props.onChange(e.target.name, e.target.value, e);
    }
  }

  render() {
    return _react2.default.createElement(
      'form',
      {
        className: this.blockName('form', []),
        action: this.props.action,
        method: this.props.method,
        name: this.props.name,
        target: this.props.target,
        onChange: this.onChange,
        onSubmit: this.props.onSubmit
      },
      _react2.default.createElement('input', { type: 'submit', style: { display: 'none' } }),
      this.props.children
    );
  }
}
exports.default = Form;
module.exports = exports['default'];