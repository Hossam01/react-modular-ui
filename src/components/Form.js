import React from 'react';
import AbstractComponent from './AbstractComponent';

export default class Form extends AbstractComponent {
  onChange(e) {
    if (typeof this.props.onChange === 'function') {
      this.props.onChange(e.target.name, e.target.value, e);
    }
  }

  render() {
    return (
      <form
        className={this.getBlockName('form', [])}
        action={this.props.action}
        method={this.props.method}
        name={this.props.name}
        target={this.props.target}
        onChange={this.onChange}
        onSubmit={this.props.onSubmit}
      >
        <input type="submit" style={{display: 'none'}} />
        {this.props.children}
      </form>
    )
  }
}
