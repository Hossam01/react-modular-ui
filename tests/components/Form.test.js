import React from 'react';
import renderer from 'react-test-renderer';
import Form from '../../src/components/Form';

describe('Form', () => {
  it('should render Form', () => {
    const tree = renderer.create(
      <Form>Something</Form>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
