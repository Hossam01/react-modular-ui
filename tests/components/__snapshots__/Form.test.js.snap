// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render Form 1`] = `
<form
  action={undefined}
  className="c-form"
  method={undefined}
  name={undefined}
  onChange={[Function]}
  onSubmit={undefined}
  target={undefined}
>
  <input
    style={
      Object {
        "display": "none",
      }
    }
    type="submit"
  />
  Something
</form>
`;
